src/Uno.Toolkit.UITest/Controls/SafeArea/Given_SafeArea.cs

```diff
@@ -223,36 +225,140 @@ public void When_Inside_Modal()
+		[ActivePlatforms(Platform.Android, Platform.iOS)]
+		public void When_SoftInput_Scroll_Into_View(string constraint, string mode)
+		{
+			const string TopTextBox = "TopTextBox";
+			const string BottomTextBox = "BottomTextBox";
+			const string SpacerBorder = "SpacerBorder";
+			var isSoft = string.Equals("Soft", constraint, StringComparison.OrdinalIgnoreCase);
+
+			NavigateToNestedSample("SafeArea_SoftInput_Scroll");
+
+			App.WaitForElement(SpacerBorder, timeoutMessage: $"Timed out waiting for element: {SpacerBorder}");
+
+			var keyboardRect = GetKeyboardRect();
+
+			App.FastTap($"{constraint}ConstraintMode");
+			App.FastTap($"{mode}Mode");
+
+			App.FastTap(TopTextBox);
+
+			App.Wait(TimeSpan.FromSeconds(1));
+
+			// Make sure the keyboard has opened and the focused TextBox is visible above the keyboard
+			Assert.Greater(keyboardRect.Height, 0);
+
+			if (isSoft)
+			{
+				var topRect = GetRectangle(TopTextBox);
+				var bottomRect = GetRectangle(BottomTextBox);
+
+				Assert.False(topRect.IsEmpty);
+				Assert.False(bottomRect.IsEmpty);
+
+				// The focused TextBox should be visible above the keyboard
+				Assert.AreEqual(bottomRect.Bottom, keyboardRect.Top);
+			}
+			else
+			{
+				var topRect = GetRectangle(TopTextBox);
+				var bottomRect = GetRectangle(BottomTextBox);
+
+				Assert.False(topRect.IsEmpty);
+
+				PlatformHelpers.On(
+					iOS: () => Assert.Greater(bottomRect.Top, keyboardRect.Top),
+					Android: () => Assert.True(bottomRect.IsEmpty)
+				);
+			}
+
+			App.DismissKeyboard();
+
+			App.FastTap(BottomTextBox);
+
+			App.Wait(TimeSpan.FromSeconds(1));
+
+			if (isSoft)
+			{
+				var topRect = GetRectangle(TopTextBox);
+				var bottomRect = GetRectangle(BottomTextBox);
+
+				Assert.False(topRect.IsEmpty);
+				Assert.False(bottomRect.IsEmpty);
+
+				Assert.Less(topRect.Bottom, keyboardRect.Top);
+				Assert.AreEqual(bottomRect.Bottom, keyboardRect.Top);
+			}
+			else
+			{
+				App.WaitForNoElement(TopTextBox, timeoutMessage: $"Timed out waiting for no element: {TopTextBox}");
+				var bottomRect = GetRectangle(BottomTextBox);
+
+				Assert.False(bottomRect.IsEmpty);
+				Assert.AreEqual(bottomRect.Bottom, keyboardRect.Top);
+			}
+
+			Rectangle GetKeyboardRect()
+			{
+				var initialBottom = App.GetPhysicalRect(BottomTextBox).Bottom;
+
+				App.FastTap($"SoftConstraintMode");
+				App.FastTap(TopTextBox);
+				App.Wait(TimeSpan.FromSeconds(1));
+
+				var newBottom = App.GetPhysicalRect(BottomTextBox).Bottom;
+				var keyboardHeight = initialBottom - newBottom;
+				var keyboardTop = initialBottom - keyboardHeight;
+				var windowRect = App.GetPhysicalScreenDimensions();
+
+				App.DismissKeyboard();
+
+				return new AppRect(0, keyboardTop, windowRect.Width, keyboardHeight).ToRectangle();
+			}
+		}
+
+		private Rectangle GetRectangle(string marked)
+		{
+			var rect = App.Marked(marked).FirstResult()?.Rect;
+			return rect != null ? App.ToPhysicalRect(rect).ToRectangle() : Rectangle.Empty;
+		}

```

Comment:
why not streamlining all this and introduce something like:
```cs
// static class AppExtensions
public static Rect? GetPhysicalRect(this IApp app, string elementName)
{
	var rect = app.WaitForElementWithMessage(elementName).FirstResult()?.Rect;
	var scaled = app.ToPhysicalRect(rect);
	
	return scaled;
}
public static Rectangle? GetPhysicalRectangle(this IApp app, string elementName) => app.GetPhysicalRect(elementName)?.ToRectangle();

// ...
app.GetPhysicalRectangle("asd") ?? default;
```

Reply:
I don't want to sit and wait for the WaitForElementWithMessage. In the case that I am using this, I don't want an exception to be thrown because the element isn't found. It's also valid if the element wasn't found, in that case I want an empty Rectangle.

I could set a shorter timeout using this streamlined method and wrap it in a try-catch perhaps